src/client/components/react/renderer.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react'
+import Story from '../fresco/fusion-story'
+import { func } from 'prop-types'
+class Renderer extends Component {
+  constructor () {
+    super()
+    this.state = {
+      comtent: () => { // hello
+      }
+    }
+  }
+  componentDidMount () {
+    let { content } = this.props
+    let story = new Story()
+    this.story = story
+    story.registerFactory('content', content)
+    story.setData({
+      id: 'main'
+    })
+  }
+  static getDerivedStateFromProps ({ content }) {
+    return {
+      content
+    }
+  }
+  componentDidUpdate () {
+    let { content } = this.state
```

Comment:
@siawo  Why are we using content from state instead of props? They are the same right? If we used the content in props, we could also probably get rid of `getDerivedStateFromProps` as well.